# EDA

In [39]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [40]:
df = pd.read_csv("items.csv")
df

mobile_deliverytype  ... mobile_rating
0                        NaN  ...           4.0
1    FREE Delivery by Amazon  ...           3.0
2                        NaN  ...           2.0
3    FREE Delivery by Amazon  ...           1.0
4                        NaN  ...           4.4
..                       ...  ...           ...
920                      NaN  ...           4.2
921  FREE Delivery by Amazon  ...           4.1
922                      NaN  ...           4.4
923  FREE Delivery by Amazon  ...           4.2
924                      NaN  ...           4.3

[925 rows x 4 columns]

In [41]:
# Checking the missing values
df.isnull().sum()

mobile_deliverytype    481
mobile_name              0
mobile_price             0
mobile_rating            0
dtype: int64

In [42]:
#fill No Free delivery it it is null
df.mobile_deliverytype.fillna("No Free Delivery", inplace = True)
df

mobile_deliverytype  ... mobile_rating
0           No Free Delivery  ...           4.0
1    FREE Delivery by Amazon  ...           3.0
2           No Free Delivery  ...           2.0
3    FREE Delivery by Amazon  ...           1.0
4           No Free Delivery  ...           4.4
..                       ...  ...           ...
920         No Free Delivery  ...           4.2
921  FREE Delivery by Amazon  ...           4.1
922         No Free Delivery  ...           4.4
923  FREE Delivery by Amazon  ...           4.2
924         No Free Delivery  ...           4.3

[925 rows x 4 columns]

In [43]:
df.mobile_price.describe()

count        925
unique        14
top       10,999
freq         148
Name: mobile_price, dtype: object

In [44]:
df = df[['mobile_deliverytype','mobile_name','mobile_price','mobile_rating']]

In [36]:
df.dtypes

mobile_deliverytype     object
mobile_name             object
mobile_price            object
mobile_rating          float64
dtype: object

In [45]:
#convert price and rating to numeric
df['mobile_price'] = df['mobile_price'].str.replace(",","").astype(int)
df

mobile_deliverytype  ... mobile_rating
0           No Free Delivery  ...           4.0
1    FREE Delivery by Amazon  ...           3.0
2           No Free Delivery  ...           2.0
3    FREE Delivery by Amazon  ...           1.0
4           No Free Delivery  ...           4.4
..                       ...  ...           ...
920         No Free Delivery  ...           4.2
921  FREE Delivery by Amazon  ...           4.1
922         No Free Delivery  ...           4.4
923  FREE Delivery by Amazon  ...           4.2
924         No Free Delivery  ...           4.3

[925 rows x 4 columns]

In [46]:
df.dtypes

mobile_deliverytype     object
mobile_name             object
mobile_price             int64
mobile_rating          float64
dtype: object

In [63]:
samsung_df = df[df['mobile_name'].str.contains("Samsung")]
redmi_df = df[df['mobile_name'].str.contains("Redmi")]
samsung_df

mobile_deliverytype  ... mobile_rating
0           No Free Delivery  ...           4.0
2           No Free Delivery  ...           2.0
7    FREE Delivery by Amazon  ...           3.7
8           No Free Delivery  ...           3.7
13   FREE Delivery by Amazon  ...           3.7
..                       ...  ...           ...
916         No Free Delivery  ...           4.1
917  FREE Delivery by Amazon  ...           4.3
919  FREE Delivery by Amazon  ...           4.0
921  FREE Delivery by Amazon  ...           4.1
924         No Free Delivery  ...           4.3

[370 rows x 4 columns]

In [61]:
#between manufactures
lessthan5k = df[df['mobile_price'] < 5000]
bet5kand10k = df[df['mobile_price'] >= 5000][df['mobile_price'] < 10000]
morethan10k = df[df['mobile_price'] >= 10000]
print(bet5kand10k)
print(lessthan5k)
print(morethan10k)

         mobile_deliverytype  ... mobile_rating
5    FREE Delivery by Amazon  ...           3.7
12          No Free Delivery  ...           4.2
14          No Free Delivery  ...           4.1
15   FREE Delivery by Amazon  ...           4.2
17   FREE Delivery by Amazon  ...           4.3
..                       ...  ...           ...
912         No Free Delivery  ...           4.2
914         No Free Delivery  ...           4.1
915  FREE Delivery by Amazon  ...           4.2
917  FREE Delivery by Amazon  ...           4.3
920         No Free Delivery  ...           4.2

[222 rows x 4 columns]
Empty DataFrame
Columns: [mobile_deliverytype, mobile_name, mobile_price, mobile_rating]
Index: []
         mobile_deliverytype  ... mobile_rating
0           No Free Delivery  ...           4.0
1    FREE Delivery by Amazon  ...           3.0
2           No Free Delivery  ...           2.0
3    FREE Delivery by Amazon  ...           1.0
4           No Free Delivery  ...           4.4
..           

# Inference


There are No mobile less than 5k <br>
There are 222 mobiles between 5k and 10k <br>
There are 703 mobiles more than 10k

In [64]:
#within  manufacturer Samsung
lessthan5k = samsung_df[samsung_df['mobile_price'] < 5000]
bet5kand10k = samsung_df[samsung_df['mobile_price'] >= 5000][df['mobile_price'] < 10000]
morethan10k = samsung_df[samsung_df['mobile_price'] >= 10000]
print(bet5kand10k)
print(lessthan5k)
print(morethan10k)

         mobile_deliverytype  ... mobile_rating
17   FREE Delivery by Amazon  ...           4.3
42   FREE Delivery by Amazon  ...           4.3
67   FREE Delivery by Amazon  ...           4.3
92   FREE Delivery by Amazon  ...           4.3
117  FREE Delivery by Amazon  ...           4.3
142  FREE Delivery by Amazon  ...           4.3
167  FREE Delivery by Amazon  ...           4.3
192  FREE Delivery by Amazon  ...           4.3
217  FREE Delivery by Amazon  ...           4.3
242  FREE Delivery by Amazon  ...           4.3
267  FREE Delivery by Amazon  ...           4.3
292  FREE Delivery by Amazon  ...           4.3
317  FREE Delivery by Amazon  ...           4.3
342  FREE Delivery by Amazon  ...           4.3
367  FREE Delivery by Amazon  ...           4.3
392  FREE Delivery by Amazon  ...           4.3
417  FREE Delivery by Amazon  ...           4.3
442  FREE Delivery by Amazon  ...           4.3
467  FREE Delivery by Amazon  ...           4.3
492  FREE Delivery by Amazon  ...       

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


# Inference

There are No mobile less than 5k <br>
There are 37 mobiles between 5k and 10k <br>
There are 33 mobiles more than 10k



In [65]:
#within  manufacturer Redmi
lessthan5k = redmi_df[redmi_df['mobile_price'] < 5000]
bet5kand10k = redmi_df[redmi_df['mobile_price'] >= 5000][df['mobile_price'] < 10000]
morethan10k = redmi_df[redmi_df['mobile_price'] >= 10000]
print(bet5kand10k)
print(lessthan5k)
print(morethan10k)

         mobile_deliverytype  ... mobile_rating
5    FREE Delivery by Amazon  ...           3.7
12          No Free Delivery  ...           4.2
14          No Free Delivery  ...           4.1
15   FREE Delivery by Amazon  ...           4.2
20          No Free Delivery  ...           4.2
..                       ...  ...           ...
905  FREE Delivery by Amazon  ...           3.7
912         No Free Delivery  ...           4.2
914         No Free Delivery  ...           4.1
915  FREE Delivery by Amazon  ...           4.2
920         No Free Delivery  ...           4.2

[185 rows x 4 columns]
Empty DataFrame
Columns: [mobile_deliverytype, mobile_name, mobile_price, mobile_rating]
Index: []
         mobile_deliverytype  ... mobile_rating
1    FREE Delivery by Amazon  ...           3.0
3    FREE Delivery by Amazon  ...           1.0
4           No Free Delivery  ...           4.4
6           No Free Delivery  ...           4.2
9    FREE Delivery by Amazon  ...           4.1
..           

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


# Inference

There are No mobile less than 5k <br>
There are 185 mobiles between 5k and 10k <br>
There are 370 mobiles more than 10k

In [69]:
# 10 top  rated samsung mobiles
samsung_df.sort_values(by='mobile_rating', ascending=False).head(10)

mobile_deliverytype  ... mobile_rating
924         No Free Delivery  ...           4.3
592  FREE Delivery by Amazon  ...           4.3
517  FREE Delivery by Amazon  ...           4.3
524         No Free Delivery  ...           4.3
199         No Free Delivery  ...           4.3
542  FREE Delivery by Amazon  ...           4.3
192  FREE Delivery by Amazon  ...           4.3
549         No Free Delivery  ...           4.3
567  FREE Delivery by Amazon  ...           4.3
574         No Free Delivery  ...           4.3

[10 rows x 4 columns]

**the above line of codes display 10 top rated samsung mobiles**

In [70]:
# 10 top  rated redmi mobiles
redmi_df.sort_values(by='mobile_rating', ascending=False).head(10)

mobile_deliverytype  ... mobile_rating
604    No Free Delivery  ...           4.4
304    No Free Delivery  ...           4.4
772    No Free Delivery  ...           4.4
672    No Free Delivery  ...           4.4
779    No Free Delivery  ...           4.4
104    No Free Delivery  ...           4.4
179    No Free Delivery  ...           4.4
479    No Free Delivery  ...           4.4
97     No Free Delivery  ...           4.4
472    No Free Delivery  ...           4.4

[10 rows x 4 columns]

**the above line of codes display 10 top rated redmi mobiles**